### PR TITLE
Add two assertions to interacts with database

### PR DESCRIPTION
**Pull request by MohammadAlhallaq**
Hello!

This PR adds the assertDatabaseIsEmpty, and assertDatabaseIsNotEmpty to the InteractsWithDatabase trait.

This will allow using:

`$this->assertDatabaseIsEmpty($table);`
`$this->assertDatabaseIsNotEmpty($table);`

instead of relying on assertDatabaseCount to check if a table is empty or not:

`$this->assertDatabaseCount($table, 0);`

### CODE REVIEW

File: src/Illuminate/Testing/Constraints/emptyDatabase.php [hunk 1]
```diff
@@ -0,0 +1,64 @@
+<?php
+
+
+namespace Illuminate\Testing\Constraints;
+
+use Illuminate\Database\Connection;
+use PHPUnit\Framework\Constraint\Constraint;
+use ReflectionClass;
+
+class emptyDatabase extends Constraint
```
Comment (jmarcher) [inline src/Illuminate/Testing/Constraints/emptyDatabase.php:10]: Filename should start with a capital letter

File: src/Illuminate/Testing/Constraints/notEmptyDatabase.php [hunk 2]
```diff
@@ -0,0 +1,64 @@
+<?php
+
+
+namespace Illuminate\Testing\Constraints;
+
+use Illuminate\Database\Connection;
+use PHPUnit\Framework\Constraint\Constraint;
+use ReflectionClass;
+
+class notEmptyDatabase extends Constraint
```
Comment (jmarcher) [inline src/Illuminate/Testing/Constraints/notEmptyDatabase.php:10]: Filename should start with a capital letter

File: src/Illuminate/Testing/Constraints/notEmptyDatabase.php [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    /**
+     * Get the description of the failure.
+     *
+     * @param string $table
+     * @return string
+     */
+    public function failureDescription($table): string
+    {
+        return sprintf(
+            'the table [%s] is empty.', $table
```
Comment (jmarcher) [inline src/Illuminate/Testing/Constraints/notEmptyDatabase.php:50]: maybe a better message: 'table [%s] is empty, expected to not be empty'

File: src/Illuminate/Testing/Constraints/emptyDatabase.php [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    /**
+     * Get the description of the failure.
+     *
+     * @param string $table
+     * @return string
+     */
+    public function failureDescription($table): string
+    {
+        return sprintf(
+            'the table [%s] is not empty.', $table
```
Comment (jmarcher) [inline src/Illuminate/Testing/Constraints/emptyDatabase.php:50]: maybe a better message: 'table [%s] is not empty, expected to be empty'

File: src/Illuminate/Foundation/Testing/Concerns/InteractsWithDatabase.php [hunk 5]
```diff
@@ -8,7 +8,9 @@
 use Illuminate\Support\Arr;
 use Illuminate\Support\Facades\DB;
 use Illuminate\Testing\Constraints\CountInDatabase;
+use Illuminate\Testing\Constraints\emptyDatabase;
```
Comment (jmarcher) [inline src/Illuminate/Foundation/Testing/Concerns/InteractsWithDatabase.php:11]: The use also needs to be fixed

File: src/Illuminate/Foundation/Testing/Concerns/InteractsWithDatabase.php [hunk 6]
```diff
@@ -8,7 +8,9 @@
...
 use Illuminate\Testing\Constraints\HasInDatabase;
+use Illuminate\Testing\Constraints\notEmptyDatabase;
```
Comment (jmarcher) [inline src/Illuminate/Foundation/Testing/Concerns/InteractsWithDatabase.php:13]: The use also needs to be fixed

File: src/Illuminate/Foundation/Testing/Concerns/InteractsWithDatabase.php (hunk 5 above)
Comment (MohammadAlhallaq) [reply src/Illuminate/Foundation/Testing/Concerns/InteractsWithDatabase.php]: thought they were refactored after I re-imported the class with the new name!

**Comment by taylorotwell** [comment]
Thanks for your pull request to Laravel!

Unfortunately, I'm going to delay merging this code for now. To preserve our ability to adequately maintain the framework, we need to be very careful regarding the amount of code we include. 

If possible, please consider releasing your code as a package so that the community can still take advantage of your contributions! 

If you feel absolutely certain that this code corrects a bug in the framework, please "**@**" mention me in a follow-up comment with further explanation so that GitHub will send me a notification of your response.

**Pull request closed without merging**